### PR TITLE
Ensure cmake honours CMAKE_CXX_STANDARD arg

### PR DESCRIPTION
**Pull request by g-217**
Ensure `cmake` honours `-DCMAKE_CXX_STANDARD=<CPP VERSION>`.

### CODE REVIEW

File: .github/workflows/adobe_source_libraries.yml [hunk 1]
```diff
@@ -94,7 +94,7 @@ jobs:
         run: |
           call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
           mkdir ..\build
-          cmake -S. -B../build -GNinja -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD=23 -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake -DBOOST_TEST_DYN_LINK=TRUE
+          cmake -S. -B../build -GNinja -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD=17 -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake -DBOOST_TEST_DYN_LINK=TRUE
```
Comment (g-217) [inline .github/workflows/adobe_source_libraries.yml:97]: Working on C++20 support for VS. [VS document](https://learn.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-170) does not mention C++23 compilation flag.

File: .github/workflows/adobe_source_libraries.yml [hunk 2]
```diff
@@ -79,7 +79,7 @@ jobs:
         shell: bash
         run: |
           mkdir ../build
-          cmake -S. -B../build -GNinja -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD=23
+          cmake -S. -B../build -GNinja -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD=20
```
Comment (g-217) [inline .github/workflows/adobe_source_libraries.yml:82]: Need higher version of clang for C++23 support.

**Review by sean-parent — COMMENTED** [review]
Changes requested.

File: .github/workflows/adobe_source_libraries.yml (hunk 2 above)
Comment (g-217) [reply .github/workflows/adobe_source_libraries.yml]: Reverted to C++17.

**Comment by sean-parent** [comment]
LGTM!

**Pull request merged**